### PR TITLE
a11y(grid-list): Add grid list to accessibility demo page

### PR DESCRIPTION
**Pull request by tinayuangao**
Remove the default roles in grid list and grid tile

### CODE REVIEW

File: src/demo-app/a11y/grid-list/grid-list-a11y.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    </md-grid-tile>
+  </md-grid-list>
+</section>
+
+<section>
+  <h2>Grid list with header and footer</h2>
+  <md-grid-list cols="3" rowHeight="200px">
+    <md-grid-tile *ngFor="let dog of dogs">
+      <md-grid-tile-header>{{dog.name}}</md-grid-tile-header>
+      <img [alt]="dog.name" src="https://material.angularjs.org/material2_assets/ngconf/{{dog.name}}.png">
```
Comment (tinayuangao) [inline src/demo-app/a11y/grid-list/grid-list-a11y.html]: I think `role="listitem"` is okay to contain images. The description in w3c https://www.w3.org/TR/wai-aria/roles#list says it' document structure with non-interactive elements. It doesn't say it cannot contain images. 

I'm okay to remove the default roles in grid list and grid tile, as users may want to use them as interactive elements.

File: src/demo-app/a11y/grid-list/grid-list-a11y.html [hunk 2]
```diff
@@ -0,0 +1,42 @@
+<section>
+  <h2>Fixed-height grid list</h2>
+  <md-grid-list [cols]="fixedCols" [rowHeight]="fixedRowHeight">
```
Comment (jelbourn) [inline src/demo-app/a11y/grid-list/grid-list-a11y.html]: With the list roles removed from the components, should we add them to the demos for these cases that are non-interactive content?

File: src/demo-app/a11y/grid-list/grid-list-a11y.html [hunk 3]
```diff
@@ -0,0 +1,42 @@
+<section>
+  <h2>Fixed-height grid list</h2>
```
Comment (jelbourn) [inline src/demo-app/a11y/grid-list/grid-list-a11y.html]: Make this header "Types of coffee (fix-height grid-list)"?

File: src/demo-app/a11y/grid-list/grid-list-a11y.html [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  <md-grid-list [cols]="fixedCols" [rowHeight]="fixedRowHeight">
+    <md-grid-tile *ngFor="let tile of tiles" [colspan]="tile.cols" [rowspan]="tile.rows"
+                  [style.background]="tile.color">
+      {{tile.text}}
+    </md-grid-tile>
+  </md-grid-list>
+</section>
+
+<section>
+  <h2>Ratio-height grid list</h2>
```
Comment (jelbourn) [inline src/demo-app/a11y/grid-list/grid-list-a11y.html]: Make this header "Types of coffee (fix-height grid-list)"?

File: src/demo-app/a11y/grid-list/grid-list-a11y.html [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  <md-grid-list cols="2" rowHeight="fit" [gutterSize]="ratioGutter"
+                [style.height]="fitListHeight">
+    <md-grid-tile *ngFor="let tile of tiles" [style.background]="'#F1EBBA'">
+      {{tile.text}}
+    </md-grid-tile>
+  </md-grid-list>
+</section>
+
+<section>
+  <h2>Grid list with header and footer</h2>
```
Comment (jelbourn) [inline src/demo-app/a11y/grid-list/grid-list-a11y.html]: Make this header "Angular team dogs (grid-list with tile headers and footers)"?

File: src/demo-app/a11y/grid-list/grid-list-a11y.html [hunk 6]
```diff
@@ -1,41 +1,41 @@
 <section>
-  <h2>Fixed-height grid list</h2>
-  <md-grid-list [cols]="fixedCols" [rowHeight]="fixedRowHeight">
+  <h2>Types of coffee (fix-height grid-list)</h2>
```
Comment (jelbourn) [inline src/demo-app/a11y/grid-list/grid-list-a11y.html]: nit: "fix" -> "fixed"

**Pull request merged**